### DISCUSSION TITLE
Proxy with ONVIF back channel output

### DISCUSSION THREAD

**Discussion by RogerHardiman** [bluenviron/gortsplib#596] [Questions]
Hi
UPDATED:
I'm trying to make a RTSP Proxy which adds ONVIF backchannel audio to an existing RTSP stream.
The goal is to run this on a Raspberry Pi (with my RPOS ONVIF code) and send backchannel audio from CCTV software to the Proxy, which can pass the audio to the Pi audio hardware
```
RTSP source --(video)---------->  GoRTSPLib Proxy ------------(Video)--------------->  CCTV Software (viewing)

Raspberry Pi audio <---(G711)---- GoRTSPLib Proxy <-----(backchannel audio)----------  CCTV Software (microphone)
```

I've made changes to gortsp lib and examples/proxy
  [https://github.com/rogerhardiman/gortsplib](https://github.com/rogerhardiman/gortsplib)

Currently I can receive RTP audio from 'examples/client-play-backchannel' and can see RTP packets arriving in the Proxy example.
(in server.go)
If I open the steam from the proxy with ffplay, it causes gortsplib to crash

Please can someone check if I'm going about this the right way as I'd like to be able to contribute my changes back into the gortsplib project.

Many thanks
Roger Hardiman

**Comment by aler9** [comment]
Hello, reading and proxying back channels has been implemented in #777. Examples are available here:

https://github.com/bluenviron/gortsplib/blob/main/examples/server-play-backchannel/main.go
https://github.com/bluenviron/gortsplib/blob/main/examples/proxy-backchannel/main.go
